importer/v8/importer.go

```diff
@@ -178,10 +181,18 @@ func (i *Importer) processDML(scanner *bufio.Reader) error {
 			return nil
 		}
 		if strings.HasPrefix(line, "# CONTEXT-DATABASE:") {
-			i.database = strings.TrimSpace(strings.Split(line, ":")[1])
+			i.dmlDatabase = strings.TrimSpace(strings.Split(line, ":")[1])
```

Comment:
Is there a reason why you don't just call `i.batchWrite()` here before changing the database? It seems like there would be less of a code change if the batch were just flushed whenever the context database or retention policy is changed. It looks like `batchWrite()` is pretty safe to call multiple times. The only caveat is that, for some reason, this section of code:

```go
i.batch = i.batch[:0]
```

This is in `batchAccumulator` rather than inside of `batchWrite`. If you move that piece of code to the end of `batchWrite` and remove it from `batchAccumulator`, you can just call `batchWrite` whenever you want to flush the writes.

Reply:
You're right, thanks. That was much easier. I rebased and pushed.

Not sure how to write a test with more than 5000 points without pasting in a gigantic string of json. Open to suggestions.